sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/broker/SempBasicAuthClientExecutor.java

```diff
@@ -0,0 +1,171 @@
+  }
+
+  private static String createQueueEndpoint(String messageVpn) {
+    return String.format("/config/msgVpns/%s/queues", messageVpn);
+  }
+
+  private static String subscriptionEndpoint(String messageVpn, String queueName) {
+    return String.format("/config/msgVpns/%s/queues/%s/subscriptions", messageVpn, queueName);
+  }
+
+  BrokerResponse getQueueResponse(String queueName) throws IOException {
+    String queryUrl = getQueueEndpoint(messageVpn, queueName);
+    HttpResponse response = executeGet(new GenericUrl(baseUrl + queryUrl));
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  BrokerResponse createQueueResponse(String queueName) throws IOException {
+    String queryUrl = createQueueEndpoint(messageVpn);
+    ImmutableMap<String, Object> params =
+        ImmutableMap.<String, Object>builder()
+            .put("accessType", "non-exclusive")
+            .put("queueName", queueName)
+            .put("owner", username)
+            .put("permission", "consume")
+            .put("ingressEnabled", true)
+            .put("egressEnabled", true)
+            .build();
+
+    HttpResponse response = executePost(new GenericUrl(baseUrl + queryUrl), params);
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  BrokerResponse createSubscriptionResponse(String queueName, String topicName) throws IOException {
+    String queryUrl = subscriptionEndpoint(messageVpn, queueName);
+
+    ImmutableMap<String, Object> params =
+        ImmutableMap.<String, Object>builder()
+            .put("subscriptionTopic", topicName)
+            .put("queueName", queueName)
+            .build();
+    HttpResponse response = executePost(new GenericUrl(baseUrl + queryUrl), params);
+    return BrokerResponse.fromHttpResponse(response);
+  }
+
+  private HttpResponse executeGet(GenericUrl url) throws IOException {
+    HttpRequest request = requestFactory.buildGetRequest(url);
+    return execute(request);
+  }
+
+  private HttpResponse executePost(GenericUrl url, ImmutableMap<String, Object> parameters)
+      throws IOException {
+    HttpContent content = new JsonHttpContent(GsonFactory.getDefaultInstance(), parameters);
+    HttpRequest request = requestFactory.buildPostRequest(url, content);
+    return execute(request);
+  }
+
+  private HttpResponse execute(HttpRequest request) throws IOException {
+    request.setNumberOfRetries(2);
+    HttpHeaders httpHeaders = new HttpHeaders();
+    boolean authFromCookie = COOKIE_MANAGER.getCookieStore().getCookies().size() > 0;
+    if (authFromCookie) {
+      setCookiesFromCookieManager(httpHeaders);
+      request.setHeaders(httpHeaders);
+    } else {
+      httpHeaders.setBasicAuthentication(username, password);
+      request.setHeaders(httpHeaders);
+    }
+
+    HttpResponse response;
+    try {
+      response = request.execute();
+    } catch (HttpResponseException e) {
+      if (authFromCookie && e.getStatusCode() == 401) {
```

Comment:
is this (make a request and check for 401) the only way to refresh token?

Reply:
The Solace [documentation](https://docs.solace.com/Admin/SEMP/SEMP-Security.htm) lacks guidance on detecting token expiration. Typically, tokens include an `expires_in` field for calculating refresh times. However, Solace seems to omit this information. Currently, the only way to check token validity is by sending a request and looking for a 401 Unauthorized response, which would then trigger token refresh.